payjoin/src/receive/mod.rs

```diff
@@ -791,7 +803,37 @@ impl PayjoinProposal {
             self.payjoin_psbt.inputs[i].final_script_sig = None;
             self.payjoin_psbt.inputs[i].final_script_witness = None;
         }
-        Ok(self.payjoin_psbt)
+        Ok(PayjoinProposal {
+            payjoin_psbt: self.payjoin_psbt,
+            owned_vouts: self.owned_vouts,
+            params: self.params,
+        })
+    }
+
+    pub fn finalize_proposal(
+        mut self,
+        wallet_process_psbt: impl Fn(&Psbt) -> Result<std::string::String, Error>,
+        min_feerate_sat_per_vb: Option<u64>,
+    ) -> Result<PayjoinProposal, RequestError> {
+        let payjoin_proposal_psbt: &Psbt = self.apply_fee(min_feerate_sat_per_vb)?;
+        let payjoin_proposal_psbt = match wallet_process_psbt(payjoin_proposal_psbt) {
+            Ok(p) => p,
+            Err(_) => {
+                return Err(RequestError::from(InternalRequestError::PsbtFinalizingError(
+                    "wallet_process_psbt failed",
+                )));
+            }
+        };
+        let payjoin_proposal_psbt = match Psbt::from_str(&payjoin_proposal_psbt.to_string()) {
+            Ok(p) => p,
+            Err(e) => return Err(RequestError::from(InternalRequestError::PsbtParseError(e))),
+        };
+
+        let payjoin_proposal = match self.prepare_psbt(payjoin_proposal_psbt) {
+            Ok(p) => p,
+            Err(e) => return Err(RequestError::from(e)),
+        };
```

Comment:
`.map_error` or even a new `to_request_error` might turn this into something less verbose

e.g. something like

```rs
let payjoin_proposal = self.prepare_psbt(payjoin_proposal_psbt).map_err(RequestError::from)?

/// or maybe it's .map_err(|e| e.into)? i'm not sure
```